pyk/src/pyk/kastManip.py

```diff
@@ -509,23 +514,21 @@ def setCell(constrainedTerm, cellVariable, cellValue):
     return mlAnd([substitute(config, subst), constraint])
 
 
-def removeUselessConstraints(constrainedTerm, keepVars=None):
-    (state, constraint) = split_config_and_constraints(constrainedTerm)
-    constraints = flatten_label('#And', constraint)
-    usedVars = collectFreeVars(state)
+def remove_useless_constraints(cterm: CTerm, keepVars=None) -> CTerm:
```

Comment:
Also rename all the variables in the method with the new scheme while you're at it.